c_src/sasl_auth.c

```diff
@@ -467,6 +491,8 @@ static ERL_NIF_TERM sasl_srv_new(ErlNifEnv* env, int UNUSED(argc), const ERL_NIF
     if (state->principal == NULL) {
         return ERROR_TUPLE(env, ATOM_OOM);
     }
+    // server never need this
+    state->user = NULL;
```

Comment:
below, there is a risk of memleak if you set it to NULL without free it while the `state->user` is not NULL.
If we are sure it must be NULL, just have assertion here.

Reply:
this is to initialized it as `NULL`.
otherwise it's a random address, segfault when trying to release.

Reply:
enif_free does nothing when it's NULL.

Reply:
oh, I miss read the diff. sorry